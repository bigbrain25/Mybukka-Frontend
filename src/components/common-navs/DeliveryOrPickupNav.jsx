import React from 'react';

import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import setDeliveryMode from './actionCreators/setDeliveryMode';

import './deliveryorpickupnav.scss';

const DeliveryOrPickupNav = ({ mode, setDeliveryModeAction, delivery }) => (
  <div className="btn-group" role="group">
    <button
      type="button"
      className={`btn-delivery ${
        mode === 'delivery' ? 'delivery-mode-active' : ''
      }`}
      onClick={() => setDeliveryModeAction('delivery')}
    >
      delivery
    </button>
    {!delivery &&
    <button
      type="button"
      className={`btn-pickup ${
        mode === 'pickup' ? 'delivery-mode-active' : ''
      }`}
      onClick={() => setDeliveryModeAction('pickup')}
    >
      pickup
    </button>
    }
  </div>
);

const mapStateToProps = ({ deliveryModeReducer: { mode } }) => ({
  mode
});

export default connect(
  mapStateToProps,
  { setDeliveryModeAction: setDeliveryMode }
)(DeliveryOrPickupNav);

DeliveryOrPickupNav.defaultProps = {
  mode: 'delivery',
  delivery: false
};

DeliveryOrPickupNav.propTypes = {
  mode: PropTypes.string,
  delivery: PropTypes.bool,
  setDeliveryModeAction: PropTypes.func.isRequired
};
