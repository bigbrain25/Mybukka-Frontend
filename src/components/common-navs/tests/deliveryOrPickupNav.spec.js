import React from 'react';

import DeliveryOrPickupNav from '../DeliveryOrPickupNav';

describe('DeliveryOrPickupNav component', () => {
  const wrapper = shallow(<DeliveryOrPickupNav />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
