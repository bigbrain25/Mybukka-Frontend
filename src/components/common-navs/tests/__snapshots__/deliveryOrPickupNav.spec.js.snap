// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryOrPickupNav component renders properly 1`] = `ShallowWrapper {}`;
